File: src/main/scala/us/zoom/data/dfence/providers/snowflake/models/opaque/ObjectType.scala
package us.zoom.data.dfence.providers.snowflake.models.opaque

import us.zoom.data.dfence.providers.snowflake.grant.builder.SnowflakeObjectType

opaque type ObjectType = String

object ObjectType:
  def apply(snowflakeObjectType: SnowflakeObjectType): ObjectType =
    snowflakeObjectType.getObjectType()

File: src/main/java/us/zoom/data/dfence/providers/snowflake/revoke/models/wrappers/ObjectType.java
package us.zoom.data.dfence.providers.snowflake.revoke.models.wrappers;

import us.zoom.data.dfence.providers.snowflake.grant.builder.SnowflakeObjectType;

public record ObjectType(String value) {
  public static ObjectType apply(SnowflakeObjectType snowflakeObjectType) {
    return new ObjectType(snowflakeObjectType.getObjectType());
  }
}
